src/types/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+add_subdirectory(tests)
+
+# Find Bison and Flex
+find_package(BISON REQUIRED)
+find_package(FLEX REQUIRED)
+find_package(Abstraction REQUIRED)
+
+# Create directory where Bison and Flex generated sources will be placed.
+set(GENERATED_SOURCE_DIR "${CMAKE_BINARY_DIR}/src/types/src")
+file(MAKE_DIRECTORY "${GENERATED_SOURCE_DIR}")
+
+set(PARSER_BASENAME "idl.parser.c")
+set(PARSER_PATH "${GENERATED_SOURCE_DIR}/${PARSER_BASENAME}")
+set(LEXER_BASENAME "idl.lexer.c")
+set(LEXER_PATH "${GENERATED_SOURCE_DIR}/${LEXER_BASENAME}")
+
+bison_target( parser "src/idl.y" "${PARSER_PATH}")
+flex_target(
+  lexer "src/idl.l" "${LEXER_PATH}"
+  COMPILE_FLAGS
+    "--header-file=${GENERATED_SOURCE_DIR}/idl.lexer.h")
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+set(CMAKE_C_FLAGS -fms-extensions)
+set_source_files_properties("${PARSER_PATH}" PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion")
+set_source_files_properties("${LEXER_PATH}" PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion -Wno-unused-parameter -Wno-unused-function")
+endif()
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+set(CMAKE_C_FLAGS "-fms-extensions -Wno-microsoft-anon-tag")
+set_source_files_properties("${PARSER_PATH}" PROPERTIES COMPILE_FLAGS "-Wno-conversion -Wno-sign-conversion")
```

Comment:
Why is the sign conversion warning suppressed? Is this something that we cannot work around in another fashion?

Reply:
These warnings are in the code generated by bison and flex. I have added one define ("#define YYTYPE_INT16 int") to yy_decl.h to remove one of the warnings in idl.parser.c. The other warning, and also most of the warnings in idl.lexer.c need explicit casting to be added to the generated code, for which there is no work-around as far as I could find.